internal/testrunner/runners/system/runner.go

```diff
@@ -247,26 +262,32 @@ func (r *runner) run() ([]testrunner.TestResult, error) {
 
 		var multiErr multierror.Error
 		for _, hit := range results.Hits.Hits {
-			if hit.Source.Error != nil {
-				multiErr = append(multiErr, errors.New(hit.Source.Error.Message))
+			if message, err := hit.Source.GetValue("error.message"); err != common.ErrKeyNotFound {
+				multiErr = append(multiErr, errors.New(message.(string)))
+				continue
+			}
+
+			err := fieldsValidator.ValidateDocumentMap(hit.Source)
+			if err != nil {
+				multiErr = append(multiErr, err)
 			}
 		}
 
 		if len(multiErr) > 0 {
-			return false, errors.Wrapf(multiErr, "one or more errors found in documents stored in %s data stream",
-				dataStream)
+			return false, testerrors.ErrTestCaseFailed{
+				Reason:  fmt.Sprintf("one or more errors found in documents stored in %s data stream", dataStream),
+				Details: multiErr.Error(),
+			}
 		}
 		return true, nil
 	}, 2*time.Minute)
-
 	if err != nil {
-		return resultsWith(result, errors.Wrap(err, "could not check for expected data in data stream"))
+		return resultsWith(result, err)
```

Comment:
Any reason to review the `errors.Wrap` here?

Reply:
It is like that because I changed the implementation of `resultsWith` function:

```go
	resultsWith := func(tr testrunner.TestResult, err error) ([]testrunner.TestResult, error) {
		tr.TimeElapsed = time.Now().Sub(startTime)
		if err == nil {
			return []testrunner.TestResult{tr}, nil
		}

		if tcf, ok := err.(testerrors.ErrTestCaseFailed); ok {
			tr.FailureMsg = tcf.Reason
			tr.FailureDetails = tcf.Details
			return []testrunner.TestResult{tr}, nil
		}

		tr.ErrorMsg = err.Error()
		return []testrunner.TestResult{tr}, err
	}
```

You can say, that I'm doing type assertion: `err.(testerrors.ErrTestCaseFailed)`.

Reply:
Ah, I see now. Thanks for pointing me to that.